icefoganalytics/elcc-data-management
# ELCC-7: Make Admin and Quality section mirror data input in Spaces

Fixes ELCC-7

# Context

The user enters data into the Spaces section, and those numbers should carry-over into the Admin and Quality sections.  The user should still be able to edit the Admin and Quality numbers.

To give more background info on that, Spaces and Admin can go over the number of licensed spots, however we will only pay up to the spots amount for Quality.

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/1c3b53dc-8d4a-4c28-a8af-cde6bca728ad)

# Implementation

Add custom mutation that propagates any change in section 1 to sections 2 and 3. This might be hard to maintain in the future, but at least it will be easy to remove. In the future, I'll see if I can write some tests to make the purpose of this code easier to understand.

Kill some dead authorization code.
Save auth token in local storage in development mode to avoid having to log back in every time you save a file and the app rebuilds.
Use the standard money formatting library that `Intl.NumberFormat`.

# Screenshots

![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/578da21b-665f-40d4-963c-b50c3d538df3)

# Testing Instructions

1. Boot the app via `dev up`

2. Check that the app complies, and that you can log in at http://localhost:8080.
3. Go to the /dashboard.
4. Click on the child care centre widget (graducation hat)
5. Click on a daycare.
6. Click on the “view details” tab on the far left widget thing.
7. Then click the “Worksheets” tab, and click on “add worksheets for 2023/24”
8. Wait for the notification “fiscal year added”.
9. Reload the page if needed. You should now see
![image](https://github.com/icefoganalytics/elcc-data-management/assets/23045206/ef806868-216a-4a78-af3e-643410e8dbfc)
10. Edit any field in section 1, and after tabbing out of the field, check that the value has propagated to sections 2 and 3.
> Note that this is a one way binding, and editing section 1 after changing section 2 will overwrite the value in section 2.
